### PR TITLE
libgcrypt: fix build on Apple Silicon

### PR DESCRIPTION
**Pull request by BrettDong**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`libgcrypt` fails to build on Apple Silicon at `make check` step. Log message reveals that the process `random`, one of the checks in `make check`, was killed by the OS. System log indicates that the process is killed because of invalid signature:
random

The reason is that after `make` and before `make check`, a workaround procedure in Formula modified the binary `random` and broke its original code signature, so the executable was refused to run. The problem is solved by adding a step of creating an ad-hoc code signature.

### CODE REVIEW

File: Formula/libgcrypt.rb [hunk 1]
```diff
@@ -41,6 +41,12 @@ def install
       MachO::Tools.change_install_name("#{buildpath}/tests/.libs/random",
                                        "#{lib}/libgcrypt.20.dylib",
                                        "#{buildpath}/src/.libs/libgcrypt.20.dylib")
+
+      # The above operation modified tests/.libs/random and
+      # broke the validity of its code signature. Have to
+      # re-sign the binary so that it can be authorized to
+      # run on Apple Silicon.
+      system("codesign", "-s", "-", "#{buildpath}/tests/.libs/random")
```
Comment (carlocab) [inline Formula/libgcrypt.rb:49]: Should this be run only on ARM? I think `Hardware::CPU.arm?` may work.

Cf. https://github.com/Homebrew/brew/pull/9102

Comment (fxcoudert) [inline Formula/libgcrypt.rb:49]:
```suggestion
      MachO.codesign("#{buildpath}/tests/.libs/random")
```

Comment (fxcoudert) [reply Formula/libgcrypt.rb:49]: Can you try this? It should work, I think

**Review by fxcoudert — DISMISSED** [review]

File: Formula/libgcrypt.rb (hunk 1 above)
Comment (BrettDong) [reply Formula/libgcrypt.rb:49]:
```
Error: An exception occurred within a child process:
  NoMethodError: undefined method `codesign' for MachO:Module
Did you mean?  codesign!
```

Comment (BrettDong) [reply Formula/libgcrypt.rb:49]: After I change it to
```ruby
MachO.codesign!("#{buildpath}/tests/.libs/random")
```
I am getting
```
==> make
==> codesign -s - /private/tmp/libgcrypt-20201211-74077-1lpxxvf/libgcrypt-1.8.7/
Last 15 lines from /Users/brett/Library/Logs/Homebrew/libgcrypt/03.codesign:
2020-12-11 18:56:06 +0800

codesign
-s
-
/private/tmp/libgcrypt-20201211-74077-1lpxxvf/libgcrypt-1.8.7/tests/.libs/random

/private/tmp/libgcrypt-20201211-74077-1lpxxvf/libgcrypt-1.8.7/tests/.libs/random: is already signed

Do not report this issue to Homebrew/brew or Homebrew/core!
```

File: Formula/libgcrypt.rb [hunk 2]
```diff
@@ -41,6 +41,7 @@ def install
       MachO::Tools.change_install_name("#{buildpath}/tests/.libs/random",
                                        "#{lib}/libgcrypt.20.dylib",
                                        "#{buildpath}/src/.libs/libgcrypt.20.dylib")
+      MachO.codesign!("#{buildpath}/tests/.libs/random")
```
Comment (fxcoudert) [inline Formula/libgcrypt.rb:44]: @carlocab is right, here, sorry:
```suggestion
      MachO.codesign!("#{buildpath}/tests/.libs/random") if Hardware::CPU.arm?
```

**Pull request merged**